docs/src/main/paradox/google-cloud-bigquery.md

```diff
@@ -15,114 +17,177 @@ Alpakka Google Cloud BigQuery was added in Alpakka 2.0.2 in July 2020 and is mar
 
-If you want to use the built in paging implementation, or you have some specific needs you can call the raw API.
-The next example shows how you can access [dryRun](https://cloud.google.com/bigquery/query-plan-explanation) data with the raw api and helpers.
+## Load data into BigQuery
+
+The BigQuery connector enables loading data into tables via real-time streaming inserts or batch loading.
+For an overview of these strategies see the [BigQuery documentation](https://cloud.google.com/bigquery/docs/loading-data).
+
+The @scala[@apidoc[BigQuery.insertAll[In]](BigQuery$)] @java[@apidoc[BigQuery.<In>insertAll](BigQuery$)] method creates a sink that accepts batches of @scala[`Seq[In]`] @java[`List<In>`]
+(for example created via the [`batch`](https://doc.akka.io/docs/akka/current/stream/operators/Source-or-Flow/batch.html) operator) and streams them directly into a table.
+To enable/disable BigQuery’s best-effort deduplication feature use the appropriate @apidoc[InsertAllRetryPolicy$].
 
 Scala
-: @@snip [snip](/google-cloud-bigquery/src/test/scala/docs/scaladsl/GoogleBigQuerySourceDoc.scala) { #dry-run }
+: @@snip [snip](/google-cloud-bigquery/src/test/scala/docs/scaladsl/BigQueryDoc.scala) { #streaming-insert }
 
 Java
-: @@snip [snip](/google-cloud-bigquery/src/test/java/docs/javadsl/GoogleBigQuerySourceDoc.java) { #dry-run }
+: @@snip [snip](/google-cloud-bigquery/src/test/java/docs/javadsl/BigQueryDoc.java) { #streaming-insert }
 
-### Config
+As a cost-saving alternative to streaming inserts, you can also add data to a table via asynchronous load jobs.
+The @scala[@apidoc[BigQuery.insertAllAsync[In]](BigQuery$)] @java[@apidoc[BigQuery.<In>insertAllAsync](BigQuery$)] method creates a flow that starts a series of batch load jobs.
+By default, a new load job is created every minute to attempt to emulate near-real-time streaming inserts, although there is no guarantee when the job will actually run.
+The frequency with which new load jobs are created is controlled by the `alpakka.google.bigquery.load-job.per-table-quota` configuration setting.
 
-The configuration will contain the session (which includes your service-token).
+@@@warning
 
-If you create multiple requests to the same source (likely to happen) you should create a single `BigQueryConfig` instance and reuse it.
+Pending the resolution of [Google BigQuery issue 176002651](https://issuetracker.google.com/176002651), the `BigQuery.insertAllAsync` API may not work as expected.
 
-If you call multiple bigquery sources (not likely to happen) it is worth to cache the configs, so you can save a lot of unneeded authorization requests.
+As a workaround, you can use the config setting `akka.http.parsing.conflicting-content-type-header-processing-mode = first` with Akka HTTP v2.6.4 or later (not yet released at time of writing but available as a [snapshot build](https://doc.akka.io/docs/akka-http/snapshot/contributing.html#snapshots)).
```

Comment:
Update/delete with Akka http version bump.

Reply:
Here's a question: should we leave it up to users decide to set `akka.http.parsing.conflicting-content-type-header-processing-mode = first` or should we set this ourselves in the `reference.conf` so that this connector works out-of-the-box?

Reply:
Oops, nevermind! The [Akka docs](https://doc.akka.io/docs/akka/current/general/configuration.html#where-configuration-is-read-from) say
> It’s not supported to override a config property owned by one library in a reference.conf of another library